src/witan/app/data.clj

```diff
@@ -26,26 +29,43 @@
   [data-id]
   (hayt/select :data_by_data_id (hayt/where {:data_id data-id})))
 
+(defn find-data-name
+  [name]
+  (hayt/select :data_names (hayt/where {:name name})))
+
+(defn update-version-number-name
+  [name version]
+  (hayt/update :data_names (hayt/set-columns {:version version})
+                           (hayt/where {:name name})))
+
+(defn get-current-version-name
+  [name]
+  (some-> (first (c/exec (find-data-name name)))
+          :version))
+
 (defn create-data
-  [{:keys [data-id category name publisher version s3-url]} data-table]
+  [{:keys [data-id category name publisher version file-name s3-key]} data-table]
   (let [creation-time (tf/unparse (tf/formatters :date-time) (t/now))]
     (hayt/insert data-table (hayt/values :data_id data-id
                                          :category category
                                          :name name
                                          :publisher publisher
                                          :version version
-                                         :s3_url s3-url
+                                         :file_name file-name
+                                         :s3_key s3-key
                                          :created creation-time))))
 
 (defn add-data!
-  "TODO: for now manually upload and provide s3 url, but need to upload as part of this process"
-  [{:keys [data-id category name publisher version s3-url]
-    :or {data-id (uuid/random)
-         version 1}}]
-  (mapv #(c/exec (create-data {:data-id data-id
-                               :category category
-                               :name name
-                               :publisher publisher
-                               :version version
-                               :s3-url s3-url} %)) '(:data_by_data_id :data_by_category))
-  (first (c/exec (find-data-by-data-id data-id))))
+  "add data version"
+  [{:keys [data-id category name file-name publisher s3-key]
+    :or {data-id (uuid/random)}}]
+  (let [version (or (get-current-version-name name) 1)]
+    (mapv #(c/exec (create-data {:data-id data-id
```

Comment:
clojure 1.7.0 has `run!`, which can be better than `doseq` for this specific use case,